btsieve/src/main.rs

```diff
@@ -82,57 +80,42 @@ fn create_bitcoin_routes(
 
     let mut bitcoin_chain = Bitcoin::default();
 
-    let (client, network) = if let Some(settings) = settings {
-        let bitcoin_rpc_client = bitcoincore_rpc::Client::new(
-            settings.node_url.to_string(),
-            settings.authentication.into(),
-        )
-        .map_err(|e| {
-            log::debug!("failed to create bitcoincore_rpc::Client: {:?}", e);
-            Error::ConnectionError {
-                ledger: "bitcoin".to_owned(),
-            }
-        })?;
-        let blockchain_info = get_bitcoin_info(&bitcoin_rpc_client)?;
-        log::info!("Connected to Bitcoin: {:?}.", blockchain_info);
-        let network = blockchain_info
-            .chain
-            .parse::<BitcoinNetwork>()
-            .map_err(|_| Error::UnknownLedgerVersion {
-                network: blockchain_info.chain,
-                ledger: "bitcoin".to_string(),
-            })?
-            .into();
-
-        log::trace!("Setting up bitcoin routes to {:?}.", network);
-
-        log::info!("Connect BitcoinZmqListener to {}.", settings.zmq_endpoint);
+    let (block_source, network) = if let Some(settings) = settings {
+        let bitcoin_http_client = reqwest::r#async::Client::new();
```

Comment:
Why do you have to instantiate the client outside?
Could also just be done inside the constructor right? :)

Reply:
ah, yeah, it was inside before but I thought I need the `Client` outside when I integrated it into `main.rs`. Turns out It did not :D 
Thanks, good catch